### PR TITLE
Feature/86/skeleton for product items

### LINKED ISSUE #86: (SP: ) Create Skeleton for product item
When the page is loading status to fetch product data should render `Skeleton` component
For e.g
image

Just for our product card while it's loading

### PR DESCRIPTION
**Pull request by ValKoDim**
Added ProductSkeleton component, which will show as a placeholder while the actual product cards are loading. This is done for the purpose of reducing user frustration during loading time.

### CODE REVIEW

File: code/src/components/product-skeleton/ProductSkeleton.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Skeleton } from "@mui/material";
```
Comment (mhevyk) [inline code/src/components/product-skeleton/ProductSkeleton.tsx]: Use `AppSkeleton`

File: code/src/layouts/best-sellers/BestSellers.tsx [hunk 2]
```diff
@@ -14,8 +15,10 @@ const BestSellers = () => {
     size: 5
   });
 
-  //@TODO Create Skeleton for component loading
-  if (isLoading) return <AppTypography>Loading...</AppTypography>;
+  const skeletonCards = Array.from(
+    { length: products?.content?.length || 10 },
+    (_, index) => <ProductSkeleton key={index} />
+  );
```
Comment (mhevyk) [inline code/src/layouts/best-sellers/BestSellers.tsx]: I suggest creating new utility for creating some amount of skeleton

Comment (ValKoDim) [reply code/src/layouts/best-sellers/BestSellers.tsx]: Do you have any suggestions as to where to put that util function? In the utils folder or maybe inside the component itself?

File: code/src/components/product-skeleton/ProductSkeleton.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.spa-product-skeleton {
+  display: flex;
+  flex-direction: column;
+}
+.spa-product-skeleton {
+  gap: spacing(10);
+}
```
Comment (dmtrth25) [inline code/src/components/product-skeleton/ProductSkeleton.scss]: `classes` the same. create once

**Comment by dmtrth25** [comment]
Also it will be great if you pass the short video how `Skeleton` is works. Smth with low loading etc.

File: code/src/layouts/best-sellers/BestSellers.tsx (hunk 2 above)
Comment (mhevyk) [reply code/src/layouts/best-sellers/BestSellers.tsx]: I think in global `utils` folder. Maybe we can make it more generic to repeat any component any number of times

File: code/src/components/product-skeleton/ProductSkeleton.test.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    isLoading: true,
+    isSuccess: true,
+    isError: false,
+    error: null
+  });
+  renderWithProviders(<BestSellers />);
+  const productSkeleton = screen.getAllByTestId("spa-product-skeleton");
+
+  expect(productSkeleton[0]).toBeInTheDocument();
+});
```
Comment (OliaHrytsak) [inline code/src/components/product-skeleton/ProductSkeleton.test.tsx:11]: Its not a test for ProductSceleton.tsx
Its a test for BestSellers that was writing before.
If you write test for Sceleton, you should import component that you tested

And this line, of course, you can add to BestSellers.test
` const productSkeleton = screen.getAllByTestId("spa-product-skeleton");`

File: code/src/components/product-skeleton/ProductSkeleton.tsx (hunk 1 above)
Comment (ValKoDim) [reply code/src/components/product-skeleton/ProductSkeleton.tsx]: Done

File: code/src/layouts/best-sellers/BestSellers.tsx (hunk 2 above)
Comment (ValKoDim) [reply code/src/layouts/best-sellers/BestSellers.tsx]: I created a separate util function that creates skeletons.

File: code/src/components/product-skeleton/ProductSkeleton.scss (hunk 3 above)
Comment (ValKoDim) [reply code/src/components/product-skeleton/ProductSkeleton.scss]: Fixed

File: code/src/components/product-skeleton/ProductSkeleton.test.tsx (hunk 4 above)
Comment (ValKoDim) [reply code/src/components/product-skeleton/ProductSkeleton.test.tsx:11]: Refactored the test completely

**Review by OliaHrytsak — APPROVED** [review]

File: code/src/components/product-skeleton/ProductSkeleton.tsx (hunk 1 above)
Comment (mhevyk) [reply code/src/components/product-skeleton/ProductSkeleton.tsx]: Did you push these changes?

File: code/src/utils/createSkeletonCards.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import ProductSkeleton from "@/components/product-skeleton/ProductSkeleton";
+
+const createProductSkeletons = (iterations: number) => {
+  return Array.from({ length: iterations }, (_, index) => (
+    <ProductSkeleton key={index} />
+  ));
+};
+
+export default createProductSkeletons;
```
Comment (mhevyk) [inline code/src/utils/createSkeletonCards.tsx:9]: Please try to accept component property to render any kind of elements, not just `ProductSkeleton`. Also don't forget to write tests

**Review by heshvab — APPROVED** [review]

File: code/src/utils/createSkeletonCards.tsx (hunk 5 above)
Comment (ValKoDim) [reply code/src/utils/createSkeletonCards.tsx:9]: Fixed the AppSkeleton thing and created unit test for the function. I decided to keep the function working with only ProductSkeleton components for now.

**Pull request merged**